src/prfmodel/stimulus.py

```diff
@@ -249,3 +257,64 @@ def create_2d_bar_stimulus(  # noqa: PLR0913 (too many arguments)
             grid=grid,
             dimension_labels=dimension_labels,
         )
+
+
+def make_video(stimulus: Stimulus, title: str | None = None, kwargs: dict | None = None) -> str:
+    """Visualize a stimulus as a video.
+
+    Parameters
+    ----------
+    stimulus: Stimulus
+        The stimulus to visualize.
+    title : str or None, Optional.
+        Title for the video animation.
+    kwargs : dict or None, Optional.
+        Additional keyword arguments passed to :class:`matplotlib.animation.ArtistAnimation`.
+
+    Returns
+    -------
+    An HTML5 string that can be rendered as a video.
+
+    Examples
+    --------
+    >>> from IPython.display import HTML
+    >>> from prfmodel.stimulus import Stimulus, make_video
+    >>> bar_stimulus = Stimulus.create_2d_bar_stimulus(num_frames=100, width=128, height=64)
+    >>> video = make_video(bar_stimulus)
+    >>> HTML(video)
+    """
+    if kwargs is None:  # TODO: do we want defaults?
+        kwargs = {"interval": 50, "blit": True, "repeat_delay": 1000}
+
+    fig, ax = plt.subplots()
+    n_frames = stimulus.design.shape[0]
+    grid_extent = _get_grid_extent(stimulus.grid)
+    ims = []
+    for i in range(n_frames):
+        im = ax.imshow(stimulus.design[i, :, :], animated=True, extent=grid_extent, origin="lower")
+        ims.append([im])
+
+    if stimulus.dimension_labels:
+        ax.set_ylabel(stimulus.dimension_labels[0], fontsize=16)
+        ax.set_xlabel(stimulus.dimension_labels[1], fontsize=16)
+
+    if title:
+        ax.set_title(title, fontsize=20)
+
+    ani = animation.ArtistAnimation(fig, ims, **kwargs)
+    html5_video = ani.to_html5_video()
+    plt.close(fig)
+    return html5_video
```

Comment:
Should we perhaps return the `ArtistAnimation` instead? Then users have more freedom in modifying it and converting it to a format of their choice.

Reply:
Fixed as suggested